components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -784,44 +842,170 @@ private void addUserAttributesToCache(SessionDataCacheEntry sessionDataCacheEntr
      * parameters to the query component of the redirection URI using the
      * "application/x-www-form-urlencoded" format
      *
-     * @param clientId
-     * @param req
+     * @param oAuthMessage
      * @return
      * @throws OAuthSystemException
      * @throws OAuthProblemException
      */
-    private String handleOAuthAuthorizationRequest(String clientId, HttpServletRequest req)
+    private String handleOAuthAuthorizationRequest(OAuthMessage oAuthMessage)
             throws OAuthSystemException, OAuthProblemException {
 
-        OAuth2ClientValidationResponseDTO clientDTO = null;
-        String redirectUri = req.getParameter("redirect_uri");
-        String pkceChallengeCode = null;
-        String pkceChallengeMethod = null;
-        boolean isPKCESupportEnabled = EndpointUtil.getOAuth2Service().isPKCESupportEnabled();
-        if (StringUtils.isBlank(clientId)) {
-            if (log.isDebugEnabled()) {
-                log.debug("Client Id is not present in the authorization request");
+        OAuth2ClientValidationResponseDTO validationResponse = validateClient(oAuthMessage);
+
+        if (!validationResponse.isValidClient()) {
+            return EndpointUtil.getErrorPageURL(validationResponse.getErrorCode(), validationResponse.getErrorMsg(), null);
+        }
+
+        OAuthAuthzRequest oauthRequest = new CarbonOAuthAuthzRequest(oAuthMessage.getRequest());
+
+        OAuth2Parameters params = new OAuth2Parameters();
+        String redirectURI = populateOauthParameters(params, oAuthMessage, validationResponse, oauthRequest);
+        if (redirectURI != null) {
+            return redirectURI;
+        }
+
+        String prompt = oauthRequest.getParam(OAuthConstants.OAuth20Params.PROMPT);
+        params.setPrompt(prompt);
+
+        redirectURI = analyzePromptParameter(oAuthMessage, params, prompt);
+        if (redirectURI != null) {
+            return redirectURI;
+        }
+
+        String sessionDataKey = UUIDGenerator.generateUUID();
+        addDataToSessionCache(oAuthMessage, params, sessionDataKey);
+
+        try {
+            oAuthMessage.getRequest().setAttribute(FrameworkConstants.RequestParams.FLOW_STATUS, AuthenticatorFlowStatus
+                    .SUCCESS_COMPLETED);
+            oAuthMessage.getRequest().setAttribute(FrameworkConstants.SESSION_DATA_KEY, sessionDataKey);
+            return EndpointUtil.getLoginPageURL(oAuthMessage.getClientId(), sessionDataKey, oAuthMessage.isForceAuthenticate(),
+                    oAuthMessage.isPassiveAuthentication(), oauthRequest.getScopes(), oAuthMessage.getRequest().getParameterMap());
+
+        } catch (IdentityOAuth2Exception e) {
+            return handleIdentityOauth2Exception(e);
+        }
+    }
+
+    private String handleIdentityOauth2Exception(IdentityOAuth2Exception e) throws OAuthSystemException {
```

Comment:
Better to rename the method since the debug log is specific to particular instance when this exception is thrown

Reply:
done